airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/manifest.yaml

```diff
@@ -0,0 +1,522 @@
+                  schema:
+                    $schema: http://json-schema.org/schema#
+                    properties:
+                      href:
+                        type: string
+                      id:
+                        type: string
+                      name:
+                        type: string
+                      survey_id:
+                        type: number
+                      type:
+                        type: string
+                    type: object
+                retriever:
+                  type: SimpleRetriever
+                  requester:
+                    type: HttpRequester
+                    url_base: https://api.surveymonkey.com/v3/
+                    path: surveys/{{ stream_partition.survey_id }}/collectors
+                    http_method: GET
+                    request_parameters: {}
+                    request_headers: {}
+                    authenticator:
+                      type: BearerAuthenticator
+                      api_token: "{{ config['credentials']['access_token'] }}"
+                    error_handler:
+                      type: CompositeErrorHandler
+                      error_handlers:
+                        - type: DefaultErrorHandler
+                          backoff_strategies:
+                            - type: WaitTimeFromHeader
+                              header: X-Ratelimit-App-Global-Minute-Reset
+                    request_body_json: {}
+                  record_selector:
+                    type: RecordSelector
+                    extractor:
+                      type: DpathExtractor
+                      field_path:
+                        - data
+                  paginator:
+                    type: DefaultPaginator
+                    page_token_option:
+                      type: RequestPath
+                    pagination_strategy:
+                      type: CursorPagination
+                      cursor_value: '{{ response.get("links", {}).get("next", {}) }}'
+                      stop_condition: '{{ not response.get("links", {}).get("next", {}) }}'
+                  partition_router:
+                    - type: CustomPartitionRouter
+                      class_name: source_surveymonkey.components.SurveyIdPartitionRouter
+                      parent_stream_configs:
+                        - type: ParentStreamConfig
+                          parent_key: id
+                          partition_field: survey_id
+                          stream:
+                            $ref: "#/definitions/survey_ids"
+                transformations:
+                  - type: AddFields
+                    fields:
+                      - path:
+                          - survey_id
+                        value: "{{ stream_slice.survey_id }}"
+spec:
+  connection_specification:
+    $schema: http://json-schema.org/draft-07/schema#
+    type: object
+    required:
+      - start_date
+      - credentials
```

Comment:
No, we have a migration config file that handles it. 

This happened because when we extended the configuration for multi-auth for this connector, we didn't have a clear plan for migrating the configuration. To avoid breaking changes, we removed this field from the required list and handled authentication with conditions instead.

Reply:
The config migration handles it, yes but before the migration occurs, we still have configs without credentials in the database. How does the platform react to having a config that does not fit the spec outside of the configuration of the source?

Reply:
Ok! Based on @girarda, the platform does not use the spec for anything else than the configuration of the source. So this should be fine in terms of "the platform breaking because there is a mismatch between the config and the spec". That being said, it is still a breaking change and should at least be documented in the migration notes.

Reply:
The difference between the existing configuration without this field and the new specification will be compensated for when any connector command is triggered (e.g., check, read, etc.). At what stage will this become a breaking change?

Reply:
If there is a bug during the low-code migration, can we revert? The answer is only if nobody created a new source during that time which means it's probably not revertable. @girarda are we confortable taking that chance?

Reply:
It's fully reversible because we have added a new field without deleting the old one, so in the end, we will have both the old fields and the migrated ones.

Reply:
seems fine to me. 

This conversation makes me think we need a concept of "non-reversible" changes, that do not require user actions to be upgraded, but that are not safe for downgrades. Out of scope for this, obviously

Reply:
> It's fully reversible because we have added a new field without deleting the old one, so in the end, we will have both the old fields and the migrated ones.

Only for old configs, not for the ones created with the new version

Reply:
If config are created with the new version, it will have the credentials field, while it does not matter for previous versions as it will be optional.

Reply:
How will the connector authenticate them? Can you run the connector using the current prod version with a config from the low-code version and share the output here?

Reply:
Let me add some details:

Before migration in the database, there can exist two types of configurations: one formatted with a credentials list and the other without (containing only the access_token field). During the migration to a low-code environment, we include a config migration script that retrieves configurations with only the access_token field and incorporates a credentials list with the access_token into them. Therefore, if someone creates their connection with low-code and later we need to revert to the Python version for them, nothing changes except for the tracking change due to the format change in the state.

This connector is tested with two types of configurations, and only one of them will be updated. The other one will be skipped for updates in the config migration logic. The configuration in GCS SECRET_SOURCE-SURVEYMONKEY__CREDS completely reflects this.

Reply:
Ok! So basically, the credentials was already part of the current version of the connector and nothing is added so we should be able to revert. I guess having everything changed from `spec.json` to the `manifest.yaml` makes it a bit hard to scope the actual changes. Thanks for the detailed description that allowed me to align with your understanding